src/widget/findondiscogs.cpp

```diff
@@ -0,0 +1,37 @@
+#include "findondiscogs.h"
+
+#include <QMenu>
+#include <QtDebug>
+
+#include "track/track.h"
+#include "util/parented_ptr.h"
+
+namespace {
+const QString kServiceTitle = QStringLiteral("Discogs");
+
+const QString kSearchUrl = QStringLiteral(
+        "https://www.discogs.com/search/?"); // List of Query needed.
+} //namespace
+
+void FindOnDiscogs::addSubmenusForServices(QMenu* pFindOnMenu, const Track& track) {
```

Comment:
FindOnDiscogs is now a FindOnDiscogsMenu, the output of the factory.

Reply:
So instead of the `addSubmenusForServices` should I use `FindOnDiscogsMenu`? Then it won't override the FindOnWeb's method and shouldn't we define it for every service? I was a bit confused about it. Could you please help me with the last commits. So I can make this better.

Reply:
I would use FindOnDiscogsMenu

Reply:
> FindOnDiscogs is now a FindOnDiscogsMenu, the output of the factory.

Then the factory should be named FindOnWebMenuFactory right?